src/category_theory/abelian/non_preadditive.lean

```diff
@@ -60,20 +60,17 @@ variables (C : Type u) [category.{v} C]
 
 /-- We call a category `non_preadditive_abelian` if it has a zero object, kernels, cokernels, binary
     products and coproducts, and every monomorphism and every epimorphism is normal. -/
-class non_preadditive_abelian :=
+class non_preadditive_abelian extends has_zero_morphisms C, normal_mono_category C,
```

Comment:
If I'm reading correctly, the change on `has_zero_morphisms C` is orthogonal to the `_mono_` changes, right? Is there motivation for this change (not that I oppose it!)

Reply:
You can't talk about (co)kernels without having zero morphisms, so the change is necessary (and not orthogonal).

Reply:
Okay!